src/modules/commander/commander.cpp

```diff
@@ -700,7 +694,7 @@ transition_result_t arm_disarm(bool arm, orb_advert_t *mavlink_log_pub_local, co
 
 	// For HIL platforms, require that simulated sensors are connected
 	if (arm && hrt_absolute_time() > commander_boot_timestamp + INAIR_RESTART_HOLDOFF_INTERVAL &&
-		is_hil_setup(autostart_id) && status.hil_state != vehicle_status_s::HIL_STATE_ON) {
+		startup_in_hil && status.hil_state != vehicle_status_s::HIL_STATE_ON) {
```

Comment:
status.hil_state is set early in commander_thread_main based on startup_in_hil. You an drop startup_in_hil here then keep startup_in_hil within commander_thread_main instead of making it static.